cmake/Modules/FindCatch_EP.cmake

```diff
@@ -28,40 +28,56 @@
 # This module defines:
 #   - CATCH_INCLUDE_DIR, directory containing headers
 #   - CATCH2_FOUND, whether Catch has been found
-#   - The Catch::Catch imported target
+#   - TILEDB_CATCH2_SOURCES_DIR directory containing headers, lib source
+#   - The Catch2::Catch2 imported target
 
 # Include some common helper functions.
 include(TileDBCommon)
 
 # Search the path set during the superbuild for the EP.
 message(VERBOSE "searching for catch in ${TILEDB_EP_SOURCE_DIR}")
-set(CATCH_PATHS ${TILEDB_EP_SOURCE_DIR}/ep_catch/single_include)
+set(CATCH_PATHS ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_EP_SOURCE_DIR}/ep_catch/src)
 
 if (NOT TILEDB_FORCE_ALL_DEPS OR TILEDB_CATCH_EP_BUILT)
   find_path(CATCH_INCLUDE_DIR
-    NAMES catch.hpp
+    NAMES catch2/catch_all.hpp
     PATHS ${CATCH_PATHS}
-    PATH_SUFFIXES "catch2"
     ${TILEDB_DEPS_NO_DEFAULT_PATH}
   )
 endif()
 
-include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(Catch2
-  REQUIRED_VARS CATCH_INCLUDE_DIR
-)
+find_package(Catch2
+  HINTS
+    ${CATCH_PATHS}
+  )
+if(Catch2_FOUND)
+  set(CATCH_INCLUDE_DIR ${Catch2_INCLUDE_DIR})
+  set(CATCH_LIBRARIES ${Catch2_LIBRARIES})
+endif()
+
+message("CATCH_INCLUDE_DIR is \"${CATCH_INCLUDE_DIR}\", CATCH_LIBRARIES is \"${CATCH_LIBRARIES}\"")
```

Comment:
All but one of these calls to `message` should be marked `VERBOSE` so they don't become more build-junk. The "found" messages for other libraries limit themselves just to a library path.

Reply:
changes